m2cgen/interpreters/mixins.py

```diff
@@ -0,0 +1,70 @@
+import sys
+
+from m2cgen import ast
+from m2cgen.interpreters.interpreter import BaseAstToCodeInterpreter
+
+
+class BinExpressionDepthTrackingMixin(BaseAstToCodeInterpreter):
```

Comment:
For some reason I thought that mixins supposed to be outside of any particular class hierarchy. This is why they are "mixed in". In our case they are still inheriting the very base class. I'm not sure that it's that important, it just doesn't feel right :)

Reply:
Otherwise mixins have attribute `_cg` which they are not aware of what it is and IDE don't resolve methods of this attribute.